.buildkite/shared-pipeline-vars

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+# This file is `source`'d before calling `buildkite-agent pipeline upload`, and can be used
+# to set up some variables that will be interpolated in the `.yml` pipeline before uploading it.
+
+# 🎗️ If you update the image to a newer Xcode version, don't forget to also update the badge in the README.md file accordingly for consistency
```

Comment:
While at it, what about adopting a `.xcode-version` file?

Reply:
yeah, I thought about it too.

But then I thought:
 - The `.xcode-version` file doesn't exist yet, that means it's not yet used by any tooling (no use of e.g. [the `xcversion` action](https://docs.fastlane.tools/actions/xcversion/) to enforce the right Xcode version to be `xcode-select`'d before fastlane can continue). So introducing it would only serve as having a conventional place to declare the expected Xcode version for humans (developers, contributors) to look and check, but not enforced by any of our tooling (yet) in this repo
 - But on the other hand, there was already a source of truth for developers and contributors to look at to know which version of Xcode they were supposed to use to work on this project: the `README.md` has a dedicated badge to document exactly that.

So that made me think that introducing the `.xcode-version` file would mean that we'd suddenly have 2 sources of truth instead of one… 🤔 

That being said, once could argue that we already have 2 sources of truth with `shared-pipeline-vars` vs the `README.md` anyway, so moving to still 2 sources of truth but those now being `.xcode-version` (that the `shared-pipelines-vars` would read and derive the `IMAGE_ID` from) and `README.md`… would not change the status quo that much 😅 

But again… given the fact that [the team see to want to be able to use different Xcode versions (stable for final releases, Xcode beta for PRs & betas) during the period where there are Xcode betas around](https://github.com/Automattic/pocket-casts-ios/pull/2065#discussion_r1722243780), it's not clear what the `.xcode-version` file would be expected to contain in such case (even if those cases only happens during a short periods when Apple releases new Xcode betas, not all year round).

In particular, [the conditional logic implemented in #2066](https://github.com/Automattic/pocket-casts-ios/pull/2066/files) allowing to pick a different Xcode version depending on final release build or not… would not really work well with a single source of truth that would be the `.xcode-version` file…

So that's why in the end I decided not to introduce it after all 😅

Reply:
> The .xcode-version file doesn't exist yet, that means it's not yet used by any tooling

Perhaps we could use this as the main factor to not introduce it now 👍
But with some convention + additional logic, I guess we could work around the remaining points in the future.